pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -97,9 +102,14 @@ findDependency() {
 }
 
 autoPatchelfFile() {
-    local dep rpath="" toPatch="$1"
+    local dep interpreter rpath="" toPatch="$1"
+
+    if [ "$(getElfClass "$toPatch")" = "ELF32" ]; then
```

Comment:
Note that this should be determined only *after* we already know that the file is an executable, which is something I already ordered wrong in the first place btw.